### PR TITLE
👷 Fix staging deployment

### PR DESCRIPTION
**Pull request by jackreimers**
Related to #118 

Add reusable workflow to remove duplicate code from staging and production workflows.
Add variable substitution for API URL so that staging and production point to their respective APIs and are not hard coded.
Add staging frontend to allowed CORS URLs on API so that the frontend can connect.

### CODE REVIEW

File: .github/workflows/stage-build-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Build and Deploy - Staging
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
```
Comment (Harry-Ross) [inline .github/workflows/stage-build-deploy.yml:7]: What is the purpose of this deployment if it has the same trigger as the main-build-deploy.yml?

Comment (jackreimers) [reply .github/workflows/stage-build-deploy.yml:7]: That's the existing functionality (file was renamed so shows as an add in the diff)
It'll be removed in https://github.com/SSWConsulting/SSW.Rules.GPT/issues/113

File: src/WebAPI/DependencyInjection.cs [hunk 2]
```diff
@@ -39,6 +37,7 @@ string rulesGptCorsPolicy
         // TODO: Set CORS in Bicep
         var productionCorsUrls = new string[]
         {
+            "https://ashy-meadow-0a2bad900.3.azurestaticapps.net",
```
Comment (matt-goldman) [inline src/WebAPI/DependencyInjection.cs:40]: I would move these to `appsettings.[env.]json`. Don't need to do that here but probably a good idea to add this to the backlog.

File: .github/workflows/stage-build-deploy.yml (hunk 1 above)
Comment (Harry-Ross) [reply .github/workflows/stage-build-deploy.yml:7]: Do you think it should be removed then? Is it still creating a duplicate deployment?

File: src/WebAPI/DependencyInjection.cs (hunk 2 above)
Comment (matt-goldman) [reply src/WebAPI/DependencyInjection.cs:40]: I see there's already a TODO for this. Could do it in bicep too.

**Review by matt-goldman — APPROVED** [review]
LGTM

File: .github/workflows/stage-build-deploy.yml (hunk 1 above)
Comment (jackreimers) [reply .github/workflows/stage-build-deploy.yml:7]: No, it deploys to both staging and prod at the same time but they are seperate instances in azure

Comment (matt-goldman) [reply .github/workflows/stage-build-deploy.yml:7]: If anything it should come out of the prod one, not this on. As @jackreimers mentioned though there's another issue tracking this. The change here is just to stabilise these deployments and separate the functionality, which will facilitate this other change too.

File: src/WebAPI/DependencyInjection.cs (hunk 2 above)
Comment (jackreimers) [reply src/WebAPI/DependencyInjection.cs:40]: I've raised an issue for this: https://github.com/SSWConsulting/SSW.Rules.GPT/issues/120

**Pull request merged**